dns/rdata.py

```diff
@@ -560,6 +563,8 @@ def _as_ipv6_address(cls, value):
             return dns.ipv6.canonicalize(value)
         elif isinstance(value, bytes):
             return dns.ipv6.inet_ntoa(value)
+        elif isinstance(value, ipaddress.IPv6Address):
+            return str(value)
```

Comment:
I think something like:
```
return dns.ipv6.inet_ntoa(value.packed)
```
might be a better idea here, for several reasons.  First, the result of `str()` can contain a scope ID, which we don't want.  Second, the code is attempting to ensure that there's a unique representation for each address, which is why string inputs are explicitly canonicalized.  There may not be any cases where `str()` would generate a different text representation than `dns.ipv6.ntoa()`, but it's better to be safe than sorry.